src/Pages/AddNewRoom/AddNewRoom.tsx

```diff
@@ -0,0 +1,178 @@
+import { Button, MenuItem, TextField } from '@mui/material'
+import Box from '@mui/material/Box';
+import { useEffect, useState } from 'react'
+import './AddNewRoom.module.scss'
+import CloudUploadIcon from '@mui/icons-material/CloudUpload';
+import { useDispatch } from 'react-redux';
+import { FacilitiesData } from '@/Redux/Features/Facilities/FacilitiesSlice';
+import { useForm } from 'react-hook-form';
+import { ChevronRight } from '@mui/icons-material';
+import { CreateRooms } from '@/Redux/Features/Rooms/CreateRoomsSlice';
+import { Link, useNavigate } from 'react-router-dom';
+import { toast } from "react-toastify";
+const AddNewRoom = () => {
+  const {
+    register,
+    handleSubmit,
+    getValues,
+    formState: { errors },
+  } = useForm();
+
+  const navigate = useNavigate()
+
+  const required = "This Field is required"
+
+  //? ***************Get Facilities Data ***************
+  const dispatch = useDispatch();
+  const [selectData, setSelectData] = useState(null)
+
+
+  const getFacilitiesData = async () => {
+    // @ts-ignore
+    let element = await dispatch(FacilitiesData())
+    // @ts-ignore
+    setSelectData(element.payload.data.facilities)
+  }
+
+  useEffect(() => {
+    getFacilitiesData()
+  }, []);
+
+
+  //! ***************Selected Input ***************
+
+  const [Facilities, setFacilities] = useState<string[]>([])
+  const facilities = getValues("facilities")
+
+
+  //! ***************Selected Images ***************
+  const [images, setImages] = useState([]);
+
+  const handleImageChange = (event: any) => {
+    const files = Array.from(event.target.files)
+    setImages(files);
+  }
+  //? ***************Send Data***************
+
+  const sendData =async (data: any) => {
+  const roomsData = await dispatch(CreateRooms({ ...data, facilities, images }))
+    if(roomsData?.payload?.message === "Room created successfully"){
+      toast.success(" Room created successfully", {
+        autoClose: 2000,
+        theme: "colored",
+      })
+      navigate('/dashboard/rooms')
+    }else{
+      toast.error(" Room was not created successfully", {
+        autoClose: 2000,
+        theme: "colored",
+      })
+    }
+    
+  }
+
+
+  return <>
+    <Box className='formContainer' component="form" onSubmit={handleSubmit(sendData)}>
+
+      <TextField variant="filled" type="number" className='roomNumber' label="Room Number"
```

Comment:
room number should be of type text since it might include letters or other marks.